spec/unit/libraries/api_spec.rb

```diff
@@ -0,0 +1,44 @@
+# coding: utf-8
+require_relative '../../spec_helper'
+
+describe 'Nexus3::Api' do
+  let(:api_client) do
+    Nexus3::Api.new('http://localhost/sample/api', 'admin', 'admin123')
+  end
+
+  let(:repo_list) do
+    [
+      {
+        'name' => 'foo',
+        'content' => 'repository.createMavenHosted(\'foo\')',
+        'type' => 'groovy',
+      },
+      {
+        'name' => 'anonymous',
+        'content' => 'security.setAnonymousAccess(Boolean.valueOf(args))',
+        'type' => 'groovy',
+      },
+    ]
+  end
+
+  before(:each) do
+    WebMock.disable_net_connect!
+  end
+
+  describe 'list_repositories' do
+    it 'returns repos' do
+      # TODO: if we add the Authorization header, the test fails, so
```

Comment:
HTTPClient send basic auth header only after a 401 error.
If you want to always send the authentication header use the force_basic_auth property https://github.com/nahi/httpclient/blob/master/lib/httpclient.rb#L592